### DISCUSSION TITLE
Keep image size inside crop area while rotating

### DISCUSSION THREAD

**Discussion by mkosir** [ValentinH/react-easy-crop#488] [Q&A]
Hey 👋 
thanks for an awesome library 🚀

Running bit out of ideas, not sure if it's a library bug for `objectFit` set to "contain", but wondering how would be possible to achieve that, while rotating image, image size auto fits inside crop area ()?
For example in [sandbox](https://codesandbox.io/s/react-easy-crop-demo-with-cropped-output-forked-d3ghxd?file=/src/index.js) image should be shown as:
![image](https://github.com/ValentinH/react-easy-crop/assets/3073826/7176c6a6-e541-45d7-aba9-ee6aa722d5ee)

The idea behind is, that user can only rotate image (landscape/portartai by 90 degrees for example), without doing any cropping to it.

**Comment by ValentinH** [comment]
> The idea behind is, that user can only rotate image (landscape/portartai by 90 degrees for example), without doing any cropping to it.

The thing is that in order to achieve that, you would need to adjust the `aspect` property otherwise the image would still be cropped.
To be honest, if you don't need to allow the user cropping the image, then I would not use this component at all. Instead I would handle the rotation logic directly.

That being said, you can achieve what you want by having the `zoom` and `aspect` properties updated based on the current rotation. See this example: https://codesandbox.io/s/react-easy-crop-demo-with-cropped-output-forked-myt3fz

**Reply by mkosir** [reply]
> The thing is that in order to achieve that, you would need to adjust the aspect property otherwise the image would still be cropped.
To be honest, if you don't need to allow the user cropping the image, then I would not use this component at all. Instead I would handle the rotation logic directly.

Thanks for the answer!
I would like to allow users to be able just to crop or just to rotate the image (or both if they want of course). 

Thinking about the behaviour and the purpose of `object-fit: 'contain'` as from specs `The replaced content is scaled to maintain its aspect ratio while fitting within the element's content box.`
Maybe the library could calculate rotated image bounding box and scale the image to fit into crop area, wdyt?

**Reply by ValentinH** [reply]
I think it would be pretty complex to implement but why not. Would you like to give it a shot? I'll personally won't have time to give to this.

**Reply by mkosir** [reply]
Thanks, maybe I come up with something 🤞
